src/gluonnlp/embedding/embed_loader.py

```diff
@@ -0,0 +1,320 @@
+    file_path = _check_and_get_path(pretrained_name_or_dir)
+    if file_path is None:
+        raise ValueError("Cannot recognize `{}`".format(pretrained_name_or_dir))
+
+    if file_path.endswith('.npz'):
+        matrix, result = _load_embedding_npz(file_path, vocab, unknown)
+    else:
+        matrix, result = _load_embedding_txt(file_path, vocab, unknown)
+    dim = matrix.shape[-1]
+    logging.info("Pre-trained embedding dim: {}".format(dim))
+    if vocab is None:
+        return matrix, result
+    else:
+        hit_flags = result
+        total_hits = sum(hit_flags)
+        logging.info("Found {} out of {} words in the pretrained embedding.".format(total_hits, len(vocab)))
+        if total_hits != len(vocab):
+            if unk_method is None:
+                found_vectors = matrix[hit_flags]
+                mean = np.mean(found_vectors, axis=0, keepdims=True)
+                std = np.std(found_vectors, axis=0, keepdims=True)
+                unfound_vec_num = len(vocab) - total_hits
+                r_vecs = np.random.randn(unfound_vec_num, dim).astype('float32') * std + mean
+                matrix[hit_flags == False] = r_vecs
+            else:
+                unk_idxs = (hit_flags == False).nonzero()[0]
+                matrix[hit_flags == False] = unk_method(vocab.to_tokens(unk_idxs))
+
+        return matrix
+
+def get_fasttext_model(model_name_or_dir='cc.en.300'):
+    """ Load fasttext model from the binaray file
+
+    This method will load fasttext model binaray file from a given file path or remote sources,
+    and return a `fasttext` model object. See `fasttext.cc` for more usage information.
+
+    Available sources:
+    ['wiki-news-300d-1M-subword', 'crawl-300d-2M-subword', \
+     'cc.af.300', ..., 'cc.en.300', ..., 'wiki.aa', ..., 'wiki.en', ..., 'wiki.zu']
+    Detailed sources can be founded by `gluonnlp.embedding.list_sources('FastText.bin')`
+
+    Parameters
+    ----------
+    model_name_or_dir : str, default 'cc.en.300'
+        A file path for a FastText binary file or the name of the FastText model.
+        This method would first check if it is a file path.
+        If not, the method will load from cache or download.
+
+    Returns
+    -------
+    fasttext.FastText._FastText:
+        A FastText model based on `fasttext` package.
+    """
+    if os.path.exists(model_name_or_dir):
+        file_path = model_name_or_dir
+    else:
+        source = model_name_or_dir
+        root_path = os.path.expanduser(os.path.join(get_home_dir(), 'embedding'))
+        embedding_dir = os.path.join(root_path, 'fasttext')
+        if source not in C.FAST_TEXT_BIN_SHA1:
+            raise ValueError('Cannot recognize {} for the bin file'.format(source))
+        file_name, file_hash = C.FAST_TEXT_BIN_SHA1[source]
+        file_path = _get_file_path('fasttext', file_name, file_hash)
+    return fasttext.load_model(file_path)

```

Comment:
I just noticed one issue. Can we support multiprocessing with FastText embedding?

Reply:
Can you give a concrete example about multiprocessing with FastText embedding? I'm not very clear about it. Thanks!

Reply:
Basically, for example:
```
fasttext_model = load_fasttext(...)
with multiprocessing.Pool(4) as pool:
   out = pool.map(..., fasttext_model.encode(...))
```

Reply:
Do you mean that we need a function which support multiprocessing inside using fasttext.cc's APIs like your example or are you worried about if fasttext.cc's APIs could work in the multiprocessing setting?

Reply:
After offline discussion with Leo, this looks good and let me merge this in.